digdag-core/src/main/java/io/digdag/core/schedule/ScheduleExecutor.java

```diff
@@ -123,22 +132,36 @@ private void runSchedule(ScheduleControl lockedSched)
 
         try {
             StoredWorkflowDefinitionWithProject def = rm.getWorkflowDetailsById(sched.getWorkflowDefinitionId());
+
+            SessionStore ss = sessionStoreManager.getSessionStore(def.getProject().getSiteId());
+            List<StoredSessionAttemptWithSession> activeAttempts = ss.getActiveAttemptsOfWorkflow(def.getId(), 1, Optional.absent());
```

Comment:
Yeah, I was also wondering if execution should only be skipped if a preceding attempt is still running. In that case it might be better to change `getActiveAttemptsOfWorkflow` to e.g. `getActiveAttemptsOfWorkflowWithSessionTimeBefore` in order to catch manually (i.e. retry, backfill, start) started session attempts.